src/main/java/address/browser/AddressBookBrowser.java

```diff
@@ -0,0 +1,273 @@
+                            % listOfContactsDisplayed.size());
+        }
+        return rangeOfIndexToLoad;
+    }
+
+    /**
+     * Gets a list of Person objects that are already loaded in the browser cache.
+     * @param listOfContactsDisplayed A list of contacts displayed on the UI.
+     * @param rangeOfIndexToLoad A range of index of the listOfContactsDisplayed that are to be loaded into the browser
+     *                           cache.
+     * @return a list of Person objects that are already loaded in the browser cache.
+     */
+    private ArrayList<Person> getListOfAlreadyLoadedPerson(List<Person> listOfContactsDisplayed,
+                                                           Range rangeOfIndexToLoad) {
+        ArrayList<Person> listOfAlreadyLoadedPerson = new ArrayList<>();
+        pages.stream().forEach(browserTab -> {
+                if (rangeOfIndexToLoad.isWithinRange(listOfContactsDisplayed.indexOf(browserTab.getPerson()))){
+                    listOfAlreadyLoadedPerson.add(browserTab.getPerson());
+                }
+            });
+        return listOfAlreadyLoadedPerson;
+    }
+
+    /**
+     * Unload profile pages that are unneeded from the cache of the browser.
+     * @param listOfContactsDisplayed A list of contacts displayed on the UI.
+     * @param rangeOfIndexToLoad A range of index of the listOfContactsDisplayed that are to be loaded into the browser
+     *                           cache.
+     */
+    private void unloadProfilePages(List<Person> listOfContactsDisplayed, Range rangeOfIndexToLoad) {
+
+        for (BrowserTab browserTab: pages) {
+            int personIndexInFilteredPerson = listOfContactsDisplayed.indexOf(browserTab.getPerson());
+
+            if (personIndexInFilteredPerson == -1) {
+                continue;
+            }
+
+            if (!rangeOfIndexToLoad.isWithinRange(personIndexInFilteredPerson)) {
+                browserTab.unloadProfilePage();
+            }
+        }
+    }
+
+    /**
+     * Selects the tab that is used to load the profile page of the person.
+     * @param person
+     */
+    private void selectTabAssignedForPerson(Person person) {
+        pages.stream().filter(browserTab
+                -> browserTab.getPerson().equals(person)).forEach(browserTab -> selectTab(pages.indexOf(browserTab)));
+    }
+
+    /**
+     * Loads profile pages into the browser.
+     * @param listOfPersonsToBeLoaded
+     */
+    private void loadProfilePages(ArrayList<Person> listOfPersonsToBeLoaded) {
+        for (int c = 0; c < listOfPersonsToBeLoaded.size(); c++){
+            Optional<BrowserTab> notAssignedTab = pages.stream().filter(page -> !page.isAssigned()).findFirst();
+            int indexOfTab = 0;
+
+            if(notAssignedTab.isPresent()){
+                indexOfTab = pages.indexOf(notAssignedTab.get());
+            }
+
+            pages.get(indexOfTab).loadProfilePage(listOfPersonsToBeLoaded.get(c));
```

Comment:
I collected the browserTab reference instead of using intStream. But i guess it is the same thing since if I use intStream, i need to get the browserTab later also.